Server/Server.Core/Server.Core.Static/Settings.hpp

```diff
@@ -14,25 +15,49 @@ struct ParamType
     constexpr static char const* DBPassword  = "--password";
 };
 
+/**
+ * @class Settings
+ * @brief Class for collecting and storing parameters.
+ * @details Class can take a parameter, store it in a map and provide it.
+ */
 class Settings
 {
 public:
     Settings() = default;
 
-    inline Settings& SetValue(std::pair<std::string, std::string> value)
+    /**
+     * @brief Setting up parameter method.
+     * @details Sets the value of <value.second> in the map with the key <value.first>.
+     *          If a key is contained in map, a value won't be changed. But, it will register this case.
+     * @param const std::pair<std::string, std::string>& value /
+     *        In other words: pair<key, value>
+     */
+    Settings& setValue(const std::pair<std::string, std::string>& value)
     {
-        _settings.try_emplace(value.first, value.second);
+        if (!_settings.try_emplace(value.first, value.second).second)
+        {
+            Base::Logger::FileLogger::getInstance().log
+            (
+                "[Settings::setValue]: Write error! Parameter with key \"" + value.first + "\" already exists.",
```

Comment:
I used the next logic:
If you try to set a value with a used key, it wont a critical. Because the correct value is already set.
But your comment prompted me to look at this case from other angle.
It is strictly not recommended to try to set the value of the current Settings object by the used key.
Then the protocol error channel is more expected.